TheWarWithin/Priorities/ShamanRestoration.simc

```diff
@@ -18,10 +20,29 @@ actions+=/bag_of_tricks
 actions+=/surging_totem,if=talent.acid_rain
 actions+=/healing_rain,if=!moving&talent.acid_rain
 actions+=/flame_shock,cycle_targets=1,if=active_enemies<3&refreshable
-# actions+=/primordial_wave
+actions+=/primordial_wave
 actions+=/lava_burst,if=(active_enemies=1|active_enemies=2&buff.lava_surge.up)&dot.flame_shock.remains>cast_time&cooldown_react
 actions+=/earth_elemental
 actions+=/lightning_bolt,if=spell_targets.chain_lightning<2|!talent.chain_lightning
 actions+=/chain_lightning,if=spell_targets.chain_lightning>1
 actions+=/flame_shock,moving=1
 actions+=/frost_shock,moving=1
+
+actions.healing+=/downpour,if=buff.unleash_life.up|buff.downpour.up&buff.downpour.remains<2
+actions.healing+=/mana_tide_totem,if=mana.pct<85
+actions.healing+=/totemic_recall,if=prev_gcd.earthen_wall_totem
+actions.healing+=/earthen_wall_totem
+actions.healing+=/chain_heal,if=buff.high_tide.up
+actions.healing+=/run_action_list,name=unleash,if=buff.unleash_life.up
```

Comment:
`call_action_list` if you want it to return here and continue.

`,strict=1` to expedite processing.  This can be done whenever a `call/run_action_list` has criteria that MUST pass immediately, and requires some judgment.  I.e., `unleash_life` isn't going to pop `up` in a few seconds on its own.  It may be `up` now and expire and become `down`, but it will not be `down` and become `up` on its own simply by virtue of time passing.